library/modules/Textures.cpp

```diff
@@ -118,30 +160,57 @@ std::vector<TexposHandle> slice_tileset(SDL_Surface* surface, int tile_px_w, int
     return handles;
 }
 
-TexposHandle Textures::loadTexture(SDL_Surface* surface) {
+TexposHandle Textures::loadTexture(SDL_Surface* surface, bool reserved) {
     if (!surface)
         return 0; // should be some error, i guess
+    if (loading_state)
+        reserved = true; // use reserved range during loading for all textures
 
     auto handle = reinterpret_cast<uintptr_t>(surface);
     g_handle_to_surface.emplace(handle, surface);
     surface->refcount++; // prevent destruct on next FreeSurface by game
-    auto texpos = add_texture(surface);
-    g_handle_to_texpos.emplace(handle, texpos);
+
+    if (reserved && reserved_range.is_installed) {
+        auto texpos = reserved_range.get_new_texpos();
+        if (texpos != -1) {
+            insert_texture(surface, texpos);
```

Comment:
this should also decrement the refcount of the dummy texture it's replacing

Reply:
there is just one dummy texture placed all over reserved range... it has refcount = 1...
if we decrement, it will be destroyed

Reply:
I'm not entirely sure of the code path here, but this is what I'm concerned about.
On close, DF would iterate through the textures vector and call FreeSurface on each texture. Since the dummy texture appears more than once, it will get passed to FreeSurface many times. The first time it passes it to FreeSurface, the refcount would be decremented to 0 and the surface would be deleted. What happens when DF tries to pass the dummy surface to FreeSurface for the second time?

Reply:
according to SDL docs passing nullptr to `FreeSurface` is safe...

Reply:
Ok, found impl https://github.com/libsdl-org/SDL/blob/SDL2/src/video/SDL_surface.c#L1572

Reply:
The pointer will *not* be null -- it will be the dummy surface pointer that we put in the vector. The second call to FreeSurface would be operating on freed memory. Maybe the safest and simplest approach would be to set the `SDL_DONTFREE` flag on the dummy surface?

Reply:
Ok, i'll put a flag